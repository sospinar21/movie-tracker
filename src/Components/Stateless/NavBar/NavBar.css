.nav {
  -webkit-border-radius: 4px;
  background-color: rgba(255, 255, 255, 0);
  border-color: #ffc65d;
  border-radius: 4px;
  border: 1px solid #ffc65d;
  display: inline-block;
  font-size: 14px;
  line-height: 120%;
  padding: 10px 14px;
  text-decoration: none;
  transition: background-color 300ms ease;
  color: #ffc65d;
  cursor: pointer; }

.nav:hover {
  background-color: rgba(255, 255, 255, 0.3);
  color: #FFF; }

.selected,
.selected:hover {
  background-color: #ffc65d;
  color: black; }

.header-container {
  background-color: black;
  box-shadow: 2px 0px 20px #130606;
  margin-bottom: 2vh;
  width: 100vw;
  display: flex; }

.header-container h1 {
  color: red;
  width: 50%;
  text-align: right;
  line-height: 2;
  text-shadow: 1px 1px #1f1f25; }

header {
  display: flex;
  justify-content: space-around;
  padding: 5vh;
  width: 39%; }
