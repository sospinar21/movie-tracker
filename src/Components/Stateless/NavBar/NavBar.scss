
.nav {
	display: inline-block;
	padding: 10px 14px;
	border: 1px solid #FFF;
	text-decoration: none;
	font-size: 14px;
	line-height: 120%;
	background-color: rgba(255,255,255, 0);
	-webkit-border-radius: 4px;
	border-radius: 4px;
	transition: background-color 300ms ease;
  cursor: pointer;
  color: #ffc65d;
	border-color: #ffc65d;
}

.nav:hover {
	background-color: rgba(255,255,255, 0.3);
  color: #FFF;
  
}

header {
  width: 100vw;
  padding: 5vh;
  display: flex;
  justify-content: space-evenly;
  background-color: black;
}
