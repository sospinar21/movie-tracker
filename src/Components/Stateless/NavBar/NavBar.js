import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css';
import { connect } from 'react-redux';

const NavBar = (props) => {
  console.log(props.user)
  if (!props.user.name) {
    return (
      <header>
        <NavLink exact to='/' className='nav' activeClassName='selected'>Home</NavLink>      
        <NavLink exact to='/login' className='nav' activeClassName='selected'>Log In</NavLink>
        <NavLink exact to='/signup' className='nav' activeClassName='selected'>Sign Up</NavLink>  
        <NavLink exact to='/favorites' className='nav' activeClassName='selected'>Favorites</NavLink>
      </header>                       
    )   
  } else {
    return (
      <header>
        <NavLink exact to='/' className='nav' activeClassName='selected'>Home</NavLink>
        <button className='nav'>Sign Out</button>
        <NavLink exact to='/favorites' className='nav' activeClassName='selected'>Favorites</NavLink>         
      </header>
    )
  }
}

const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(NavBar);