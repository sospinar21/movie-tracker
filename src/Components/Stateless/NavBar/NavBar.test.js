import React from 'react';
import { NavBar, mapStateToProps, mapDispatchToProps } from './NavBar';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('NavBar', () => {

  let mockProps;
  let navBar;

  beforeEach(() => {

    mockProps = {
      user: {name: 'Steph'},
      logout: jest.fn()
    };

    navBar = shallow(<NavBar {...mockProps} />);
  });

  describe('component', () => {

    it('matches snapshot', () => {
      navBar = shallow(<NavBar {...mockProps} />, { disableLifecycleMethods: true });
      expect(navBar).toMatchSnapshot();
    });

    it('returns a different menu if the user is logged in', () => {

      navBar = shallow(<NavBar {...mockProps} />);

      const userMenu = navBar.find('div');
      expect(userMenu.hasClass('user-logged-in')).toBe(true);

    });

    it('returns a different menu if the user is logged out', () => {

      const mockProps = {user: {name: ''}}
      navBar = shallow(<NavBar  {...mockProps} />);

      const userMenu = navBar.find('div');
      expect(userMenu.hasClass('user-logged-in')).toBe(false);

    });
  });

  describe('mapStateToProps', () => {

    it('returns an object with user', () => {
      
      const mockState = {
        user: {name: 'Tony'}
      };
  
      const mappedProps = mapStateToProps(mockState);
  
      expect(mappedProps).toEqual(mockState);
    });

  });

  describe('mapStateToProps', () => {
    
    it('should call dispatch ', () => {

      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'LOGOUT'
      };

      mappedProps.logout();
      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    })
  });
  
});