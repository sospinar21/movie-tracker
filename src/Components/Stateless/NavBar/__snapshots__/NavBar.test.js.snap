// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    logout={[Function]}
    user={
        Object {
            "name": "Steph",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <NavLink
                    activeClassName="selected"
                    ariaCurrent="true"
                    className="nav"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="selected"
                    ariaCurrent="true"
                    className="nav"
                    to="/favorites"
          >
                    Favorites
          </NavLink>
          <button
                    className="nav"
                    onClick={[Function]}
          >
                    Sign Out
          </button>
</header>,
        <h1>
          Welcome 
          Steph
</h1>,
      ],
      "className": "header-container user-logged-in",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="selected"
              ariaCurrent="true"
              className="nav"
              exact={true}
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="selected"
              ariaCurrent="true"
              className="nav"
              to="/favorites"
>
              Favorites
</NavLink>,
            <button
              className="nav"
              onClick={[Function]}
>
              Sign Out
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "ariaCurrent": "true",
              "children": "Favorites",
              "className": "nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Out",
              "className": "nav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome ",
            "Steph",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Welcome ",
          "Steph",
        ],
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <NavLink
                        activeClassName="selected"
                        ariaCurrent="true"
                        className="nav"
                        exact={true}
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="selected"
                        ariaCurrent="true"
                        className="nav"
                        to="/favorites"
            >
                        Favorites
            </NavLink>
            <button
                        className="nav"
                        onClick={[Function]}
            >
                        Sign Out
            </button>
</header>,
          <h1>
            Welcome 
            Steph
</h1>,
        ],
        "className": "header-container user-logged-in",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="selected"
                ariaCurrent="true"
                className="nav"
                exact={true}
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="selected"
                ariaCurrent="true"
                className="nav"
                to="/favorites"
>
                Favorites
</NavLink>,
              <button
                className="nav"
                onClick={[Function]}
>
                Sign Out
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "ariaCurrent": "true",
                "children": "Favorites",
                "className": "nav",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "className": "nav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome ",
              "Steph",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Welcome ",
            "Steph",
          ],
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
